### PR TITLE
TCIOS-809030: Update TrueHost sample app

### PR DESCRIPTION
**Pull request by sreedeepkesav**

### CODE REVIEW

File: example/TrueSDKSample/QATitleSelectionTableViewController.swift [hunk 1]
```diff
@@ -0,0 +1,76 @@
+//
+//  QATitleSelectionTableViewController.swift
```
Comment (elnaqah) [inline example/TrueSDKSample/QATitleSelectionTableViewController.swift]: what does QA Means?

File: example/TrueSDKSample/QATitleSelectionTableViewController.swift [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+class QATitleSelectionTableViewController: UITableViewController {
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+    }
+
+    // MARK: - Table view data source
+
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        // #warning Incomplete implementation, return the number of sections
```
Comment (elnaqah) [inline example/TrueSDKSample/QATitleSelectionTableViewController.swift]: remove warning

File: example/TrueSDKSample/QATitleSelectionTableViewController.swift [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        return 1
+    }
+
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return titles.count
+    }
+    
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = UITableViewCell()
```
Comment (elnaqah) [inline example/TrueSDKSample/QATitleSelectionTableViewController.swift]: why not dequeuing ?

File: example/TrueSDKSample/QATitleSelectionTableViewController.swift [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    return ["Default",
+            "Login to",
+            "Sign up with",
+            "Sign in to",
+            "Verify with",
+            "Get started with",
+            "Register with"]
+}
+
+class QATitleSelectionTableViewController: UITableViewController {
```
Comment (elnaqah) [inline example/TrueSDKSample/QATitleSelectionTableViewController.swift]: final

File: example/TrueSDKSample/QATitleSelectionTableViewController.swift [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        return cell
+    }
+    
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        let title = titleType(for: indexPath.row)
+        postNotificationForTitleChange(with: title, titleName: titles[indexPath.row])
+    }
+}
+
+extension QATitleSelectionTableViewController {
```
Comment (elnaqah) [inline example/TrueSDKSample/QATitleSelectionTableViewController.swift]: why extension?

File: example/TrueSDKSample/QATitleSelectionTableViewController.swift [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+        let alert = UIAlertController(title: "",
+                                      message: "Title set to: \(titleName)",
+            preferredStyle: .alert)
+        alert.addAction(UIAlertAction(title: "OK",
+                                      style: .default,
+                                      handler: { action in
+            self.dismiss(animated: true, completion: nil)
+        }))
+        
+        self.present(alert, animated: true, completion: nil)
```
Comment (elnaqah) [inline example/TrueSDKSample/QATitleSelectionTableViewController.swift]: self not needed

File: example/TrueSDKSample/HostViewController.swift [hunk 7]
```diff
@@ -173,11 +173,35 @@ class HostViewController: UIViewController, UITableViewDataSource, UITableViewDe
...
+        }
+        
+        TCTrueSDK.sharedManager().titleType = titleType
+    }
+    
+    @IBAction func openTitleSelection() {
+        let controller = QATitleSelectionTableViewController()
+        controller.title = "Select Title"
+        let navigationControlelr = UINavigationController(rootViewController: controller)
+        present(navigationControlelr, animated: true, completion: nil)
```
Comment (elnaqah) [inline example/TrueSDKSample/HostViewController.swift]: if you are presenting the viewcontroller from here why not use delegation why do you use a broadcast from the notification center? it is only observed in one place

File: example/TrueSDKSample/QATitleSelectionTableViewController.swift [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+//
+
+import UIKit
+import TrueSDK
+
+extension Notification.Name {
+    static let titleChanged = Notification.Name("loginTitleChanged")
+}
+
+private var titles : [String] {
```
Comment (elnaqah) [inline example/TrueSDKSample/QATitleSelectionTableViewController.swift]: I think you should make it a private let inside the class

File: example/TrueSDKSample/HostViewController.swift [hunk 9]
```diff
@@ -173,11 +173,35 @@ class HostViewController: UIViewController, UITableViewDataSource, UITableViewDe
         userDataHeightConstraint.constant = CGFloat(userDataModel.count) * propertyRowHeight
         
         TCTrueSDK.sharedManager().delegate = self
```
Comment (elnaqah) [inline example/TrueSDKSample/HostViewController.swift]: I know it is an old code but also why set the delegate in didSet

File: example/TrueSDKSample/HostViewController.swift [hunk 10]
```diff
@@ -173,11 +173,35 @@ class HostViewController: UIViewController, UITableViewDataSource, UITableViewDe
...
+        TCTrueSDK.sharedManager().titleType = .default
+        changeTitleTypeOnNotification()
```
[the remark this answers is not in the document]
Comment (sreedeepkesav) [reply example/TrueSDKSample/HostViewController.swift]: Moved to delegate

File: example/TrueSDKSample/AppDelegate.swift [hunk 11]
```diff
@@ -13,9 +13,8 @@ import TrueSDK
 class AppDelegate: UIResponder, UIApplicationDelegate {
 
     var window: UIWindow?
-    
 
-    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplicationLaunchOptionsKey: Any]?) -> Bool {
+    internal func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
```
[the remark this answers is not in the document]
Comment (sreedeepkesav) [reply example/TrueSDKSample/AppDelegate.swift]: It was kind of an autocomplete which i missed out. Fixed

File: example/TrueSDKSample/QATitleSelectionTableViewController.swift (hunk 1 above)
Comment (sreedeepkesav) [reply example/TrueSDKSample/QATitleSelectionTableViewController.swift]: Removed

File: example/TrueSDKSample/QATitleSelectionTableViewController.swift (hunk 3 above)
Comment (sreedeepkesav) [reply example/TrueSDKSample/QATitleSelectionTableViewController.swift]: Done

File: example/TrueSDKSample/QATitleSelectionTableViewController.swift (hunk 5 above)
Comment (sreedeepkesav) [reply example/TrueSDKSample/QATitleSelectionTableViewController.swift]: Thought of moving the notification part as a separate extension. Just for clarity

File: example/TrueSDKSample/QATitleSelectionTableViewController.swift (hunk 6 above)
Comment (sreedeepkesav) [reply example/TrueSDKSample/QATitleSelectionTableViewController.swift]: Done

File: example/TrueSDKSample/HostViewController.swift (hunk 7 above)
Comment (sreedeepkesav) [reply example/TrueSDKSample/HostViewController.swift]: Done

File: example/TrueSDKSample/QATitleSelectionTableViewController.swift (hunk 8 above)
Comment (sreedeepkesav) [reply example/TrueSDKSample/QATitleSelectionTableViewController.swift]: Done

File: example/TrueSDKSample/HostViewController.swift (hunk 9 above)
Comment (sreedeepkesav) [reply example/TrueSDKSample/HostViewController.swift]: Its in viewDidLoad. I did not get your qn

Comment (elnaqah) [reply example/TrueSDKSample/HostViewController.swift]: ah sorry wrong place

File: example/TrueSDKSample/TitleSelectionTableViewController.swift [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+//
+
+import UIKit
+import TrueSDK
+
+protocol TitleSelectionTableViewControllerDelegate: class {
+    func didSet(title: TitleType)
+}
+
+class TitleSelectionTableViewController: UITableViewController {
```
Comment (elnaqah) [inline example/TrueSDKSample/TitleSelectionTableViewController.swift]: final

**Pull request merged**